work/worker.go

```diff
@@ -0,0 +1,151 @@
+package work
+
+import (
+	"context"
+	"sync"
+
+	"github.com/filecoin-project/lotus/node/modules/helpers"
+	logging "github.com/ipfs/go-log/v2"
+	"go.uber.org/fx"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/schedule"
+)
+
+var log = logging.Logger("worker-manager")
+
+func NewWorkerManager(mctx helpers.MetricsCtx, lc fx.Lifecycle) *WorkerManager {
+	return &WorkerManager{
+		WorkerID: 1,
+		Workers:  make(map[int]Worker),
+	}
+}
+
+// WorkerManager controls Workers allowing them to be started and stopped independently of each other.
+type WorkerManager struct {
```

Comment:
Suggestion: can you combine `WorkerManager` and `schedule.Scheduler`. The advantage that Scheduler has is that it can restart tasks if they fail. It would need modifying to allow tasks to be added at any time and for them to be stopped by ID. Maybe add a status struct for each one that can hold a list of recent events such as "restarted at 11:03"

We can also get clearer about naming:

 - Task - a single capability of visor, such as extracting block data from a tipset
 - Job - a scheduled/managed set of tasks with configuration (walk is a Job, watch is a Job)
 - Scheduler - responsible for executing Jobs and ensuring they continue to run as specified

(I don't like xxxManager naming)

I think this means that you could rename `schedule.Task` to `schedule.Job` and combine your `Worker` type into Job so it gets the Start and Stop methods. Rename `schedule.TaskConfig` too. It has a name field, perhaps that could be used as the ID that you have for workers.

Reply:
> I think this means that you could rename schedule.Task to schedule.Job and combine your Worker type into Job 

Sounds good, I'll adapt this PR to do that and wire it up to watch, with walk coming in a follow-on.

> Maybe add a status struct for each one that can hold a list of recent events such as "restarted at 11:03"

I was thinking of adding Journaling to tasks -- this seems like a good starting point for that.